packages/amplication-data-service-generator/src/update-package-jsons.ts

```diff
@@ -24,9 +25,8 @@ export function updatePackageJSONs(
 }
 
 function updatePackageJSON(module: Module, update: Record<string, any>) {
-  const pkg = JSON.parse(module.code);
-
-  Object.assign(pkg, update);
+  let pkg = JSON.parse(module.code);
+  pkg = merge(pkg, update);
```

Comment:
Why did you decide to use merge instead of object.assign? 
Also, I think you should look at a solution @abrl91 implemented for YAML files that allow the user to select specific paths to be updated (using _.set)

https://github.com/amplication/amplication/blob/refactor/changes-for-db-plugins/packages/amplication-data-service-generator/src/util/prepare-yaml-file.ts

Reply:
I think that @abrl91 also used merge, merge function works recursively.